include/deepworks/nn.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <deepworks/placeholder.hpp>
+#include <deepworks/layer_info.hpp>
+#include <deepworks/call.hpp>
+
+namespace deepworks {
+
+template <typename D>
+struct BaseOp {
+    BaseOp(LayerInfo info) : m_info(std::move(info)) { }
+    Placeholder operator()(Placeholder in) {
+        Call call{m_info};
+        call.pass({in});
```

Comment:
std::move

Reply:
No, `in.shape()` is used below, better to copy `in` because it's free, instead of copying `shape`